horace_core/sqw/@sqw/private/recompute_urange.m

```diff
@@ -34,10 +34,30 @@
 pix0 = h_ave.uoffset;               % pix offset
 u_to_rlu=w.data.u_to_rlu(1:3,1:3);  % proj to rlu
 u0 = w.data.uoffset;                % proj offset
-u_q=(u_to_rlu\pix_to_rlu)*(w.data.pix.q_coordinates) + u_to_rlu\repmat((pix0(1:3)-u0(1:3)),1,npixtot);
-urange=zeros(2,4);
-urange(1,1:3)=min(u_q,[],2)';
-urange(2,1:3)=max(u_q,[],2)';
-urange(1,4)=min(w.data.pix.dE) + (pix0(4)-u0(4));
-urange(2,4)=max(w.data.pix.dE) + (pix0(4)-u0(4));
 
+min_uq = realmax*ones(size(w.data.pix.q_coordinates));
+max_uq = -realmax*ones(size(w.data.pix.q_coordinates));
+min_dE = realmax;
+max_dE = -realmax;
+while true  % loop through pages of pixel data
+    u_q = (u_to_rlu\pix_to_rlu)*(w.data.pix.q_coordinates) ...
```

Comment:
the constants here `u_to_rlu\pix_to_rlu`, `u_to_rlu\(pix0(1:3)-u0(1:3))` do not need to be recalculated on each loop, you're just calculating: `u_q = A*w.data.pix.q_coordinates + B`